### PR TITLE
Added seed parameter to randomSample (#410)

### PR DESCRIPTION
**Pull request by baffelli**
That should allow resuming cached pipelines.

### CODE REVIEW

File: src/main/groovy/nextflow/extension/RandomSampleOp.groovy [hunk 1]
```diff
@@ -21,15 +21,16 @@ class RandomSampleOp {
...
 
     private List reservoir = []
 
     private int counter
 
-    RandomSampleOp( DataflowReadChannel source, int N ) {
+    RandomSampleOp( DataflowReadChannel source, int N, Integer seed = null) {
         this.source = source
         this.N = N
+        this.rng = seed ? new Random(seed) : new Random()
```
Comment (baffelli) [inline src/main/groovy/nextflow/extension/RandomSampleOp.groovy]: Good point, I did not think of that case!

**Comment by pditommaso** [comment]
Almost there! please do not commit changes to `Const.groovy` since it's automatically modify and it only cause conflicts. Could you try to remove the ones you have pushed (a `commit --amend` and `push -f` should work).

**Comment by baffelli** [comment]
Did not pay attention. Fixed!

**Comment by pditommaso** [comment]
Great! thanks!

**Comment by pditommaso** [comment]
Oops, could you please add this the `seed` option in the [doc](https://github.com/nextflow-io/nextflow/blob/master/docs/operator.rst#randomsample)

**Pull request merged**